docs/proposals/scheduling/specify-replicas/specify-replicas.md

```diff
@@ -0,0 +1,602 @@
+
+### Non-Goals
+
+## Proposal
+
+### User Stories (Optional)
+
+#### Story 1: Specify the number of replicas per cluster scenario
+<!--
+指定每个集群的副本数场景
+当deployment第一次接入Karmada的时候, 用户希望服务的副本数分布不变, 例如:
+迁移之前的服务实数分布是: `sum=7, cluster1=2,cluster2=5`
+迁移之后的服务实数分布依然是: `sum=7, cluster1=2,cluster2=5`
+-->
+
+Specify the number of replicas per cluster scenario
+When the deployment connects to Karmada for the first time, the user hopes that the distribution of the number of replicas in the service will remain unchanged, for example:
+The service real number distribution before migration is: `sum=7, cluster1=2,cluster2=5`
+The service real number distribution after migration is still: `sum=7, cluster1=2,cluster2=5`
+
+#### Story 2: Specify the number of replicas for each Region scenario
+
+<!--
+指定每个Region的副本数场景
+用户希望服务在每个Region的副本数是确定的; 如果有扩缩容,新增或者减少的副本数均匀分摊到region的每个集群中
+例如: 
+扩容场景
+-->
+
+Users hope that the number of replicas of the service in each Region is determined; if there is expansion or contraction, the new or reduced number of replicas will be evenly distributed to each cluster in the region.
+For example:
+
+increase replicas scenario
+```
+before scheduling-->  sum: 7, cluster1(RegionA): 2, cluster2(RegionB): 1, cluster3(RegionB): 4
+increase replicas-->  sum:11,  RegionA: 3,  RegionB: 8
+After scheduling--> sum:11, cluster1(RegionA): 3, cluster2(RegionB): 3, cluster3(RegionB): 5
+```
+reduce replicas scenario
+```
+before scheduling--->  sum: 7, cluster1(RegionA): 2, cluster2(RegionB): 1, cluster3(RegionB): 4
+reduce replicas-->  sum:3,  RegionA: 1,  RegionB: 2
+After scheduling--> sum:3, cluster1(RegionA): 1, cluster2(RegionB): 0, cluster3(RegionB): 2
+```
+#### Story 3:  Specify the schedulable region list
+<!--
+指定可调度的region列表
+用户希望指定可调度的region列表和总实副本数,并不关心具体的分布; 如果有扩缩容,新增或者减少的副本数均匀分摊到region的每个集群中
+扩容场景
+-->
+
+The user wants to specify the schedulable region list and total number of replicas, and does not care about the specific distribution; if there is expansion or contraction, the new or reduced number of replicas will be evenly distributed to each cluster in the region.
+
+increase replicas scenario
+```
+before scheduling-->  sum: 7, cluster1(RegionA): 2, cluster2(RegionB): 1, cluster3(RegionB): 4
+increase replicas-->  sum:11,  schedulable Region list: RegionA,RegionB
+After scheduling--> sum:11, cluster1(RegionA): 3, cluster2(RegionB): 3, cluster3(RegionB): 5
+```
+reduce replicas scenario
+```
+before scheduling-->  sum: 7, cluster1(RegionA): 2, cluster2(RegionB): 1, cluster3(RegionB): 4
+reduce replicas-->  sum:3,  schedulable Region list: RegionA,RegionB
+After scheduling--> sum:3, cluster1(RegionA): 1, cluster2(RegionB): 0, cluster3(RegionB): 2
+```
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+#### Risk
+<!--
+在Divided场景下,指定的总副本数之和可能和deployment.spec.replicas不相等
+-->
+
+In the Divided scenario, the sum of the specified total number of replicas may not be equal to deployment.spec.replicas
```

Comment:
When would such a situation occur?

Reply:
> When would such a situation occur?

The problem may lie in the upper-layer API, so that the total number of replicas specified in PP is different from deployment.spec.replicas